features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/index/IndexSelector.java

```diff
@@ -0,0 +1,190 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.flows.elastic.index;
+
+import static org.opennms.plugins.elasticsearch.rest.index.IndexStrategy.DAILY;
+import static org.opennms.plugins.elasticsearch.rest.index.IndexStrategy.HOURLY;
+import static org.opennms.plugins.elasticsearch.rest.index.IndexStrategy.MONTHLY;
+import static org.opennms.plugins.elasticsearch.rest.index.IndexStrategy.YEARLY;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalAdjusters;
+import java.time.temporal.TemporalUnit;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+
+import org.opennms.netmgt.flows.filter.api.TimeRangeFilter;
+import org.opennms.plugins.elasticsearch.rest.index.IndexStrategy;
+import org.osgi.util.measurement.Unit;
+
+public class IndexSelector {
+
+
+    private final static ZoneId UTC = TimeZone.getTimeZone("UTC").toZoneId();
+    private final static Map<IndexStrategy, TemporalUnit> UNIT_MAP;
+
+    static {
+        UNIT_MAP = new HashMap<>();
+        UNIT_MAP.put(YEARLY, ChronoUnit.YEARS);
+        UNIT_MAP.put(MONTHLY, ChronoUnit.MONTHS);
+        UNIT_MAP.put(DAILY, ChronoUnit.DAYS);
+        UNIT_MAP.put(HOURLY, ChronoUnit.HOURS);
+
+    }
+
+    private String prefix;
+    private IndexStrategy strategy;
+    private TemporalUnit unit;
+
+    public IndexSelector(String prefix, IndexStrategy strategy) {
+        this.prefix = prefix;
+        this.strategy = strategy;
+        this.unit = UNIT_MAP.get(strategy);
+        if (unit == null) {
+            // should never happen
+            throw new UnsupportedOperationException("This is a programming mistake, please check your code!");
+        }
+    }
+
+    /**
+     * We assume that timeRange is valid:
+     * - not Null
+     * - start and end date is positive
+     * - start date <= end date
+     * <p>
+     * We need to be aware that the indexes are part of the GET URL and therefor we can't have too many, it will result
+     * in  a HTTP 400 from elasticsearch. Therefor we collapse the indexes with wildcards where applicable</p>
+     */
+    public List<String> getIndexNames(TimeRangeFilter timeRange) {
+        List<String> all = new ArrayList<>();
+        // we start one before and end one after the time range, see HZN-1281
```

Comment:
The intent of the padding here is to make sure that all flows in the range are included. I think we should opt for adding an absolute unit of time, say 2 minutes, instead of one relative to the indexing strategy (i.e. one week for weekly, one month for monthly, etc..).

Maybe we should also make this absolute time configurable via something like `maxFlowDurationMs` in `org.opennms.features.flows.persistence.elastic.cfg`.